docs/api/sql/Raster-operators.md

```diff
@@ -2482,14 +2482,16 @@ SELECT RS_Normalize(band)
 
 ### RS_NormalizeAll
 
-Introduction: Normalizes values in all bands of a raster between a given normalization range. The function maintains the data type of the raster values by ensuring that the normalized values are cast back to the original data type of each band in the raster. By default, the values are normalized to range [0, 255]. RS_NormalizeAll can take upto 6 of the following arguments.
+Introduction: Normalizes values in all bands of a raster between a given normalization range. The function maintains the data type of the raster values by ensuring that the normalized values are cast back to the original data type of each band in the raster. By default, the values are normalized to range [0, 255]. RS_NormalizeAll can take upto 7 of the following arguments.
 
 - `raster`: The raster to be normalized.
 - `minLim` and `maxLim` (Optional): The lower and upper limits of the normalization range. By default, normalization range is set to [0, 255].
-- `noDataValue` (Optional): Defines the value to be used for missing or invalid data in raster bands. By default, noDataValue is set to `maxLim`.
+- `noDataValue` (Optional): Defines the value to be used for missing or invalid data in raster bands. By default, noDataValue is set to `maxLim` and Safety mode is triggered.
 - `minValue` and `maxValue` (Optional): Optionally, specific minimum and maximum values of the input raster can be provided. If not provided, these values are computed from the raster data.
 - `normalizeAcrossBands` (Optional): A boolean flag to determine the normalization method. If set to true (default), normalization is performed across all bands based on global min and max values. If false, each band is normalized individually based on its own min and max values.
 
+A safety mode is triggered when `noDataValue` is null, setting it to `maxLim` and normalizing data values to the range [minLim, maxLim-1]. This is to avoid replacing valid data that might coincide with the new `noDataValue`.
```

Comment:
Does your current implementation allow `null` as the input of a Sedona Spark (not just sedona-common)? Do we have a test?

If yes, then the sentence here should be `when `noDataValue` is null or not given). If no, it should be `when `noDataValue` is not given`

Reply:
This is an important consideration, thanks!
Current implementation allows `noDataValue` to be set to `null` in Sedona Spark as well. Confirmed this by testing it in Sedona-spark.
Have added tests for the same under Sedona-common. Would it be better to add tests under Sedona-spark as well?

Reply:
Accordingly updated the documentation too

Reply:
@prantogg https://github.com/apache/sedona/blob/master/spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/MapAlgebra.scala#L182

InferredExpression is null safe, which means it returns null directly if one of the input is null. Please double check.

In addition, `RS_NormarlizeAll` should not be put in `MapAlgebra.scala`. It should be put in `RasterEditors.scala`.

Reply:
My bad, I misunderstood

Reply:
InferredExpression is indeed null safe. Have updated the docs accordingly and moved `RS_NormalizeAll` to RasterEditors.